tracer/src/Datadog.Trace/AppSec/ObjectExtractor.cs

```diff
@@ -161,23 +158,33 @@ private static string GetPropertyName(string fieldName)
 
         private static object ExtractType(Type itemType, object value, int depth, HashSet<object> visited)
         {
+            if (itemType == typeof(string))
```

Comment:
Referring to @tonyredondo's comment here: https://github.com/DataDog/dd-trace-dotnet/pull/4302#discussion_r1266492643
> typeof is already performant, in IL this is transformed into a ldtoken and then a GetTypeFromHandle that is really fast because is only a getter to the runtime handle field. https://source.dot.net/#System.Private.CoreLib/src/System/Type.CoreCLR.cs,97
>So I don't think caching is required here.

Also, ran some tests roughly and, although variable, results weren't so obvious 🤔 
![image](https://github.com/DataDog/dd-trace-dotnet/assets/8353486/8bc00d99-9c05-4276-a7bd-45ebcae276e8)